### ISSUE TITLE
Full 3D Model?

### ISSUE THREAD

**Issue by nar1117** [mitxela/PrecisionClock-Hardware#3]
Hey Tim - 

I noticed on your project page for the Clock you reference (with photos) a full 3D model of the clock complete with the digits, PCB, and rear-facing components.

The VRML files (.wrl) in the 3d_models directory are a good start, but I don't see a model for the entire assembled clock.

Do you have the ability to export a full model from KiCad? I can import the WRL files just fine into 3ds Max (my modeling program of choice - at my day job).

I downloaded the project and have KiCad 6.0 installed, and when I opened the project in KiCad I realized it's way over my head. I'm a 3D modeling & rendering guy, not a PCB guy... so I have no idea what I'm doing in KiCad.

Long story short... I'm designing and prototyping a 3D-printed case/enclosure for the clock, and it would be a huge bonus to have an accurate 3D model available to design around. I have the basic shapes measured and modeled, but after a few iterations of modeling and printing, I'm now wanting to be a little more precise with the pieces.

Hopefully I'll have a sleek design to show off soon! Let me know if you can export a full model of the assembled clock. Or, point me in the right direction to do it myself with KiCad.

Thanks!

**Comment by mitxela** [comment]
Exporting the full 3D model from KiCad is quite easy, just open the PCB editor (not the 3d viewer) and select File > Export > VRML. I suppose I can just drop it here: [precision-clock-3D.zip](https://github.com/mitxela/PrecisionClock-Hardware/files/10541060/precision-clock-3D.zip)

Having said that, I can't guarantee the dimensions are correct. Specifically,
- the LDR is shown straight, whereas I suggest mounting it at a right angle or bent around so it faces forwards
- the DST enable pins are shown straight but a right-angle header is provided in the kit
- the 7-segment displays were modelled on a much earlier batch, there may be small differences in their heights off the PCB. The supplier I use at the moment provides digits that are very close to 34.1mm by 24.0mm by 10.5mm, but sometimes they are taller
- the GPS module and its antenna aren't mounted on the model. If you want to minimise the space, it can be mounted completely flush to the PCB, but the antenna will still need to go somewhere
- the laser-cut parts aren't shown on the model. Unless you sand/file them down, they will have a depth of 11mm, so slightly proud of the digits, and the colon LEDs will poke forward further still, unless you mount them recessed or file the top of them. 
- the ATtiny is shown soldered directly down, I provide a DIP socket in the kit which will raise its height
- the transistors and 100n caps are missing from the model (probably a path error)
- maybe other stuff

![image](https://user-images.githubusercontent.com/19376076/215626342-60c6db68-0460-4490-93ce-5fe55bb502e4.png)

**Comment by nar1117** [comment]
Great! Thank you for the guidance. I was able to export/import pretty easily on my own. The minor differences in size are not a huge concern, and I'm test-fitting all my printed parts on my own clock. The dimensions of the digits (specifically the shape of the holes on the top & bottom of each digit) are the most important. 

My case/enclosure mounting solution is taking advantage of the trapezoidal spaces between the digits and the PCB. Most of the digits allow for a small piece to slide through from top to bottom, which will connect two pieces of the mount. In some cases, a component prevents this from being possible, but it's not a big deal. There is also a cage-like piece I've modeled to protect the components on the rear of the PCB.

I've modeled & printed a few versions using this method, and it works very well. The pieces are difficult to print due to their small size, but I think it's going to turn out nicely. It's all friction-fit with dovetail joints.
